wd.c

```diff
@@ -589,6 +592,34 @@ void wd_free_list_accels(struct uacce_dev_list *list)
 	}
 }
 
+struct uacce_dev *wd_get_accel_dev(char *alg_name)
+{
+	struct uacce_dev_list *list;
+	struct uacce_dev *dev;
+	int cpu = sched_getcpu();
+	int node = numa_node_of_cpu(cpu);
+	int dis = 1024, tmp;
+
+	list = wd_get_accel_list(alg_name);
+	if (!list)
+		return NULL;
+
+	dev = list->dev;
+
+	while (list) {
+		tmp = numa_distance(node, list->dev->numa_id);
+		if (dis > tmp && tmp > 0) {
+			dev = list->dev;
+			dis = tmp;
+		}
+		list = list->next;
+	}
+
+	wd_free_list_accels(list);
```

Comment:
I think this is wrong as you free the list, so below return dev will be NULL

Reply:
wd_free_list_accels does not free dev